crypto/src/hash.rs

```diff
@@ -53,6 +54,9 @@ macro_rules! define_hash {
             std::cmp::Ord,
             std::hash::Hash,
         )]
+        // TODO: Hash -> Array, in current shape it doesnt seem to be
+        // super effective object is still allocated on heap and clone is
+        // costful, even
```

Comment:
Still has to be evaluated, but we have this: https://github.com/simplestaking/tezedge/pull/424

Reply:
my impression is that thread safe Arc as well as heap allocation might cause noticeable overhead...